components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/dao/impl/DeviceDAOImpl.java

```diff
@@ -435,6 +435,37 @@ public EnrolmentInfo getEnrolment(DeviceIdentifier deviceId, String currentOwner
         }
     }
 
+    public int getEnrolmentByStatus(DeviceIdentifier deviceId, Status status,
```

Comment:
What's the usecase for having this method? Instead cannot we use a method to get status, given the device id of the device?